app/services/hmpps_api/prison_api/offender_api.rb

```diff
@@ -89,16 +88,12 @@ def self.get_bulk_sentence_details(booking_ids)
           client.post(route, booking_ids)
         }
 
-        data.each_with_object({}) { |record, hash|
-          next unless record.key?('bookingId')
```

Comment:
It looks like we've lost the guard clause which filtered out records that were missing a `bookingId`.

Not sure if this is important / needed, or if it was unnecessary? Was it purposefully removed?

Reply:
yes - the bookingId field is mandatory from the API.